als-common/shared/src/main/scala/org/mulesoft/amfintegration/vocabularies/integration/AlsVocabularyRegistry.scala

```diff
@@ -32,7 +32,7 @@ case class AlsVocabularyRegistry(initBases: Map[String, TermsDescription])
     mut
   }
 
-  private def index(base: String, terms: Seq[DomainElement]) = {
+  def index(base: String, terms: Seq[DomainElement]) = {
```

Comment:
this should be still private, if PatchedHover needs to index something use this more specific trait instead of VocabularyProvider (or while indexing do a match to see if it is a registry and only then index the vocabulary)

Reply:
I don't think that is necessary to keep this method private.

At the end of the day there is no real difference in calling:
`def index(base: String, terms: Seq[DomainElement])`
or creating a Vocabulary to then call:
`def index(voc: Vocabulary)`

We would be creating a Vocabulary and instantly dropping it

https://github.com/aml-org/als/blob/66f516f6f0b569a3817a67131088b019169491ce/als-common/shared/src/main/scala/org/mulesoft/amfintegration/vocabularies/integration/AlsVocabularyRegistry.scala#L22-L25

Reply:
makes sense!